### PR TITLE
Qmaps 590 error

### PR DESCRIPTION
**Pull request by Julien-laville**
create generic error handler and use it on every unhandled error (local_storage & network)

```
{"name":"erdapfel","hostname":"jlaville-XPS-15-9560","pid":10051,"req":{"method":"POST","url":"/logs","headers":{"host":"localhost:3000","connection":"keep-alive","content-length":"189","pragma":"no-cache","cache-control":"no-cache","origin":"http://localhost:3000","user-agent":"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/69.0.3497.102 Safari/537.36 Vivaldi/2.0.1309.29","content-type":"application/json;charset=UTF-8","accept":"*/*","referer":"http://localhost:3000/place/osm:node:2379542204@Mus%C3%A9e_dOrsay","accept-encoding":"gzip, deflate, br","accept-language":"fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7"},"remoteAddress":"::1","remotePort":44282},"level":30,"body":{"file":"local_store","method":"get","message":"error parsing item with key qmaps_v1_favorite_place_osm:node:2379542204","exception":"SyntaxError: Unexpected token > in JSON at position 0"},"msg":"Received client log","time":"2018-12-11T09:56:22.096Z","v":0}

``` 
error log format

### CODE REVIEW

File: src/libs/telemetry.js [hunk 1]
```diff
@@ -14,7 +15,7 @@ export default class Telemetry {
     if(event) {
       return Telemetry.send(event)
     } else {
-      console.error('Telemetry event is missing')
+      Error.display('local_store', 'getAllPois', 'error getting pois', {})
```
Comment (amatissart) [inline src/libs/telemetry.js]: copy/paste fail: this message is not correct ;)

File: src/scss/includes/panels/favorite_panel.scss [hunk 2]
```diff
@@ -114,8 +114,11 @@
   padding-left: 14px;
 }
 
-.favorite_panel__item__image.icon:before{
+.favorite_panel__item__image.icon:before {
   line-height: 28px;
+  display: block;
+  height: 38px;
+  overflow: hidden;
```
Comment (amatissart) [inline src/scss/includes/panels/favorite_panel.scss]: this is probably not supposed to be part of this PR :)

File: tests/integration/tests/poi.js [hunk 3]
```diff
@@ -245,7 +245,7 @@ afterEach(async () => {
   try {
     await clearStore(page) /* if only the above test is run page is not used */
   } catch (e) {
-    console.error(e)
+    fire('error_h', `poi clear store error ${e}`)
```
Comment (amatissart) [inline tests/integration/tests/poi.js]: why this change ? Is `error_h` still used ?

File: src/panel/error_panel.js [hunk 4]
```diff
@@ -3,9 +3,6 @@ import Panel from '../libs/panel'
 
 function Error() {
   this.message = ''
-  listen('error_h', (message = '') => {
-    this.showMessage(message)
-  })
```
Comment (amatissart) [inline src/panel/error_panel.js]: I think this error panel is not used anymore.  
Could we remove it from the main app panel ? Or delete the file completely ?

File: src/libs/ajax.js [hunk 5]
```diff
@@ -33,7 +34,7 @@ const query = (url, data, method = 'GET') => {
         try {
           jsonResponse = JSON.parse(this.response)
         } catch (e) {
-          console.error('json', this.response)
+          Error.displayOnce('ajax', 'query', 'error parsing json', e)
```
Comment (amatissart) [inline src/libs/ajax.js]: Is this error really useful ? Is that not redundant with the rejected error ?

Comment (amatissart) [reply src/libs/ajax.js]: Also, some message coudl be improved here. For example line 51 `reject(xhr.status)`. Some more context could be useful (`url`, etc.)

File: src/adapters/error.js [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+  log() {
+    Ajax.post(errorUrl, this.serialise())
+  }
+
+  serialise() {
+    return {...this, exception : this.exception.toString()}
+  }
+
+  static displayOnce(file, method, message, exception) {
```
[the remark this answers is not in the document]
Comment (Julien-laville) [reply src/adapters/error.js]: displayOnce can me rename to sendOnce (but we need to display every time for local debug) 
have you suggestion ?

File: src/panel/error_panel.js (hunk 4 above)
Comment (Julien-laville) [reply src/panel/error_panel.js]: yep ill clear it

File: src/libs/telemetry.js (hunk 1 above)
Comment (Julien-laville) [reply src/libs/telemetry.js]: oups :(

File: tests/integration/tests/poi.js (hunk 3 above)
Comment (Julien-laville) [reply tests/integration/tests/poi.js]: it come from replaceAll before a build another error handler

File: src/scss/includes/panels/favorite_panel.scss (hunk 2 above)
Comment (Julien-laville) [reply src/scss/includes/panels/favorite_panel.scss]: yep, i failled something

File: src/libs/ajax.js (hunk 5 above)
Comment (Julien-laville) [reply src/libs/ajax.js]: you are right, for now i let the caller handle the error.

Comment (Julien-laville) [reply src/libs/ajax.js]: i prefer let the callee handle the error if it need, it already know url etc.

File: src/adapters/poi/idunn_poi.js [hunk 7]
```diff
@@ -23,7 +25,7 @@ export default class IdunnPoi extends Poi {
         return
       }
       else {
-        fire('error_h', err)
+        Error.sendOnce('idunn_poi', 'poiApiLoad', 'unknown error getting idunn poi', err)
```
Comment (amatissart) [inline src/adapters/poi/idunn_poi.js]: Since ajax module does not always provide details about the error, it would be useful to add the url in the message here.

File: src/libs/i18n.js [hunk 8]
```diff
@@ -21,7 +22,7 @@ I18n.prototype.setLang = async function() {
   try {
     await AsyncFileLoader(`statics/build/javascript/message/${this.language.locale}.js`)
   } catch (e) {
-    console.error(e)
+    Error.send('i18n', 'setLang', 'error getting downloading language file', e)
```
Comment (amatissart) [inline src/libs/i18n.js]: Idem, add details about the language we tried to fetch.
